📖 Add onboarding documentation for the Release Team (#9565)
kubernetes-sigs/cluster-api: docs/release/release-team-onboarding.md

- Familiarize yourself with the Release Process:
    - Review the release [tasks document](../release/release-tasks.md) which explains the responsibilities and tasks for each role within the release team.
- Check the Release Timeline:
    - Go through the [release timeline](../release/releases) of the release cycle you are involved in (i.e checkout `release-1.6.md` if you are part of the 1.6 cycle release team) to better understand the key milestones and deadlines.

Now, let's dive into the specific onboarding notes for each sub-team below.

### Release Lead Shadows Team

- Explore CI Setup and Tools:
    - Gain some insights into how CI is setup in the Cluster API and tools used. You can check out CI Team onboarding [notes](#ci-signal-lead-shadows-team) for more details.
- Learn about Release Notes Generation:
    - Get to know how the release notes generation tools works.
- Review the onboarding notes for both the Comms and CI Lead Shadows teams:
    - Check out the onboarding notes of [Comms](#comms-lead-shadows-team) and [CI Signal](#ci-signal-lead-shadows-team) teams.

> ### Comms Lead Shadows Team

Author: @g-gaston i would be happy to know your opinion on this one and add anything I missed here

- Understand Release Process: 

>     - Familiarize yourself with project's release process and the [tool](https://github.com/kubernetes-sigs/cluster-api/tree/main/hack/tools/release) used for generating release notes.

Reviewer: ```suggestion
    - Familiarize yourself with with the project's release process.
    - Walk through the [release note generation process](../release/release-tasks.md#create-pr-for-release-notes) and try to generate notes by yourself. This is the most important process the comms team is in charge of.
    - Familiarize yourself with the release notes tool [code](https://github.com/kubernetes-sigs/cluster-api/tree/main/hack/tools/release). You'll probably need to update this code during the release cycle to cover new cases or add new features.
```

I love it. Just a minor addition :)

Author: Thanks a lot @g-gaston, added the suggestions to the notes 👍🏼

- Documentation familiarity:
    - Explore project's documentation and start learning how to update and maintain it.
- Communication channels:
    - Learn about project's primary communication channels, including Slack, mailing list and office hours. Try to understand how information is distgributed and how to use them effectively for announcements, updates and discussions related to the release cycle.

> ### CI Signal Lead Shadows Team

Author: @nawazkh i would be happy to know your opinion on this one and add anything I missed here

Reviewer 2: Looks good to me so far @furkatgofurov7 . Thanks for putting it together.

<!-- START doctoc generated TOC please keep comment here to allow auto update -->
<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->

- [Onboarding Notes for Release Team Shadows](#onboarding-notes-for-release-team-shadows)
  - [Overview](#overview)
    - [Release Lead Shadows Team](#release-lead-shadows-team)
    - [Comms Lead Shadows Team](#comms-lead-shadows-team)
    - [CI Signal Lead Shadows Team](#ci-signal-lead-shadows-team)

<!-- END doctoc generated TOC please keep comment here to allow auto update -->

# Onboarding Notes for Release Team Shadows

## Overview

> Welcome to the Cluster API Release Team Shadows onboarding documentation!

Reviewer 3: ```suggestion
Welcome to the Cluster API Release Team onboarding documentation!
```

> As a member of the Release Team and Shadow, you play a crucial role in helping to ensure project releases

Reviewer 3: ```suggestion
As a member of the Release Team, you play a crucial role in helping to ensure project releases
```

I think generally we replaced the terminology 'Shadow' with 'member of release team'

Reviewer 3: `Shadow` should be removed throughout.

Author: thanks, done

are delivered smoothly and in a timely manner. While there are specific responsibilities for each sub-team within

> the Release Team Shadows, below are some general notes that every member of the Releease Team might benefit by going 

Reviewer 3: ```suggestion
the Release Team, below are some general notes that every member of the Release Team might benefit by going 
```

through at the beginning of the cycle:

- Slack:
    - Most discussions related CAPI Release topics happens in the #cluster-api channel on the Kubernetes Slack. If you need access to the Kubernetes Slack, please visit http://slack.k8s.io/.
- Kubernetes SIG membership:
    -  Try to become an official member of the Kubernetes SIG, if possible. More information on the membership and requirements can be found [here](https://github.com/kubernetes-sigs/cluster-api/blob/main/docs/release/release-team.md#cluster-api-release-team-vs-kuberneteskubernetes-sig-membership).
- Familiarize yourself with the Release Process:
    - Review the release [tasks document](../release/release-tasks.md) which explains the responsibilities and tasks for each role within the release team.
- Check the Release Timeline:
    - Go through the [release timeline](../release/releases) of the release cycle you are involved in (i.e checkout `release-1.6.md` if you are part of the 1.6 cycle release team) to better understand the key milestones and deadlines.

Now, let's dive into the specific onboarding notes for each sub-team below.

> ### Release Lead Shadows Team

Reviewer 3: ```suggestion
### Release Lead Team
```

Reviewer 3: We should probably have a better name for this team - Release Management Team? And then the Release Lead is a single person.

Reviewer 3: Out of scope for this change though.

Author: > We should probably have a better name for this team - Release Management Team? And then the Release Lead is a single person.

I am not sure if coming up with a name at all specifically for the Release Lead Shadows is a good idea since we don't have it for other sub-teams (CI/Comms). 

 I'll leave this open for now, and if we decide to introduce a new team name we can do it in a separate issue/PR.

Reviewer 3: It's just that CI team, and comms team are clear, and we have CI leads and comms leads.

 Release lead team isn't as clear as we also have a single role which is release lead.

Author: I'll create a new issue to discuss this one, as it is not directly related to this change

- Explore CI Setup and Tools:
    - Gain some insights into how CI is setup in the Cluster API and tools used. You can check out CI Team onboarding [notes](#ci-signal-lead-shadows-team) for more details.
- Learn about Release Notes Generation:
    - Get to know how the release notes generation tools works.
- Review the onboarding notes for both the Comms and CI Lead Shadows teams:
    - Check out the onboarding notes of [Comms](#comms-lead-shadows-team) and [CI Signal](#ci-signal-lead-shadows-team) teams.

> ### Comms Lead Shadows Team

Reviewer 3: ```suggestion
### Comms Team
```

- Understand Release Process: 
    - Get to know how project's release process works.
    - Walk through the [release note generation process](../release/release-tasks.md#create-pr-for-release-notes) and try to generate notes by yourself. This is the most important process the comms team is in charge of.
    - Familiarize yourself with the release notes tool [code](https://github.com/kubernetes-sigs/cluster-api/tree/main/hack/tools/release). You'll probably need to update this code during the release cycle to cover new cases or add new features.
- Documentation familiarity:
    - Explore project's documentation and start learning how to update and maintain it.
- Communication channels:

>     - Learn about project's primary communication channels, including Slack, mailing list and office hours. Try to understand how information is distgributed and how to use them effectively for announcements, updates and discussions related to the release cycle.

Reviewer 3: ```suggestion
    - Learn about project's primary communication channels, including Slack, mailing list and office hours. Try to understand how information is distributed and how to use them effectively for announcements, updates and discussions related to the release cycle.
```

Author: 🤦🏼‍♂️ fixed

> ### CI Signal Lead Shadows Team

Reviewer 3: ```suggestion
### CI Signal Lead Team
```

Author: Removed the "Lead" as well to align with your previous suggestion.

<!-- START doctoc generated TOC please keep comment here to allow auto update -->
<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->

> **Table of Contents**  *generated with [DocToc](https://github.com/thlorenz/doctoc)*

Reviewer 4: ```suggestion
```

Author: Yeah, removed now. It was added automatically when I run doctoc script locally 🤷🏼‍♂️
